clis/tcli/src/bin/tclid/daemon.rs

```diff
@@ -0,0 +1,119 @@
+                let pid: usize = pid_str.trim().parse()?;
+                match system.process(sysinfo::Pid::from(pid)) {
+                    Some(_) => Ok(true),
+                    None => Ok(false),
+                }
+            }
+            Err(e) => match e.kind() {
+                std::io::ErrorKind::NotFound => Ok(false),
+                _ => Err(e.into()),
+            },
+        }
+    }
+
+    /// Function for conveniently printing the stderr file of the daemon.
+    /// Used by the API to quickly inform the user if something is wrong
+    /// with the daemon.
+    ///
+    /// # Returns
+    ///
+    /// The whole contents of the daemons stderr file as a String.
+    pub fn get_stderr() -> Result<String> {
+        let stderr_file = get_wd_path()?.join("tclid.err");
+
+        let stderr_string = std::fs::read_to_string(stderr_file)?;
+
+        Ok(stderr_string)
+    }
+
+    /// Forks the daemon from the API process.
+    ///
+    /// # Returns
+    ///
+    /// Current process type indicating whether it's the API or the Daemon.
+    /// This is useful because of how the fork syscall.
+    pub fn start() -> Result<ProcessType> {
+        #[cfg(unix)]
+        {
+            use std::fs::File;
+            // This crate only works on unix systems and not Windows.
+            use daemonize::Daemonize;
+
+            let tclid_wd_path = get_wd_path()?;
+
+            std::fs::create_dir_all(&tclid_wd_path)?;
+            let stdout = File::create(tclid_wd_path.join("tclid.out"))?;
+            let stderr = File::create(tclid_wd_path.join("tclid.err"))?;
+
+            // TODO Create nordvpn group if it doesn't exist.
+            let daemonize = Daemonize::new()
+                .pid_file("tclid.pid") // Every method except `new` and `start`
+                .chown_pid_file(true) // is optional, see `Daemonize` documentation
+                .working_directory(tclid_wd_path) // for default behavior.
+                .group("nordvpn") // Group name
+                .stdout(stdout) // Redirect stdout to `/tmp/daemon.out`.
+                .stderr(stderr) // Redirect stderr to `/tmp/daemon.err`.
+                .privileged_action(|| "Executed before privileges are dropped");
+
+            match daemonize.execute() {
+                daemonize::Outcome::Parent(Ok(daemonize::Parent { .. })) => Ok(ProcessType::Api),
+                daemonize::Outcome::Parent(Err(err)) => Err(err.into()),
+                daemonize::Outcome::Child(Ok(_)) => Ok(ProcessType::Daemon),
+                daemonize::Outcome::Child(Err(err)) => Err(err.into()),
+            }
+        }
+        #[cfg(not(unix))]
+        todo!("Windows support not implemented for TCLID");
+    }
+}
+
+/// Private helper function for getting the path to the daemon's working directory.
+///
+/// # Returns
+///
+/// Result containing the path of the daemons work directory.
+fn get_wd_path() -> Result<PathBuf> {
+    Ok(std::env::current_exe()?
```

Comment:
Maybe [current_dir](https://doc.rust-lang.org/std/env/fn.current_dir.html) would be better choise when `current_exe`?

Reply:
Or one more idea: [data_dir](https://docs.rs/dirs/latest/dirs/fn.data_dir.html)

Reply:
Changed to current_dir, as data_dir doesn't quite fit our use case